modules/system/traits/InteractsWithZip.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace System\Traits;
+
+use Illuminate\Support\Facades\Lang;
+use Winter\Storm\Exception\ApplicationException;
+use Winter\Storm\Filesystem\Zip;
+
+trait InteractsWithZip
+{
+    /**
+     * Extract the provided archive
+     *
+     * @throws ApplicationException if the archive failed to extract
+     */
+    public function extractArchive(string $archive, string $destination): void
+    {
+        if (!Zip::extract($archive, $destination)) {
+            throw new ApplicationException(Lang::get('system::lang.zip.extract_failed', ['file' => $archive]));
+        }
+
+        @unlink($archive);
```

Comment:
Is it expected that the original archive is always removed?

Reply:
I would expect so? Reckon it should be an optional param?

Reply:
Yes probably. If we're using the preserver then we wouldn't want to delete the copy after restoring it I would think